### DISCUSSION TITLE
Django permissions

### DISCUSSION THREAD

**Discussion by FinemechanicPub** [django-cms/django-filer#1325] [Q&A] [answered]
Does this part from the docs still stand?

> By default files can be uploaded and managed by all staff members based on the standard django model permissions.

I see Django permissions that you can set via users and groups are not functional with django-filer. And looking at the source code I can't see how they can be functional. Any suggestions on where I can be wrong?

**Comment by Droxt1** [comment, marked as answer]
By default, Django's built-in permissions system is not directly integrated with Django Filer. Django Filer uses its own permission system to control access to files and folders.

However, it is possible to customize Django Filer to use Django's built-in permissions system, if desired. This can be done by subclassing the FolderPermission and FilePermission models provided by Django Filer and modifying them to use Django's built-in User and Group models for permissions, rather than the filer.models.FolderPermission and filer.models.FilePermission models provided by Django Filer.

Here is an example of how you might subclass FolderPermission and FilePermission to use Django's built-in permissions system:


```
from django.contrib.auth.models import User, Group
from filer.models import FolderPermission, FilePermission

class CustomFolderPermission(FolderPermission):
    user = models.ForeignKey(User, on_delete=models.CASCADE)
    group = models.ForeignKey(Group, on_delete=models.CASCADE)

class CustomFilePermission(FilePermission):
    user = models.ForeignKey(User, on_delete=models.CASCADE)
    group = models.ForeignKey(Group, on_delete=models.CASCADE)
```
Then, you can specify your custom FolderPermission and FilePermission models in the Django Filer settings in your Django project's settings file:

```
FILER_FOLDER_PERMISSION_MODEL = 'myapp.CustomFolderPermission'
FILER_FILE_PERMISSION_MODEL = 'myapp.CustomFilePermission'
```
With these changes, you can use Django's built-in permissions system to control access to files and folders in Django Filer.
